controller/src/test/java/org/jboss/as/controller/operation/global/EnhancedSyntaxTestCase.java

```diff
@@ -342,6 +356,68 @@ public void testComplexAttributes() throws OperationFailedException {
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName() + "[6]");
+        Assert.assertFalse(executeForResult(op).isDefined());
+
+        // test write to non-existing element
+        op = createOperation("write-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName() + "[6].attr1");
+        op.get(VALUE).set("newvalue");
+        executeCheckNoFailure(op);
+        op = createOperation("read-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName() + "[6].attr1");
+        Assert.assertEquals("newvalue", executeForResult(op).asString());
+
+        // test add to undefined list
+        op = createOperation("undefine-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName());
+        executeCheckNoFailure(op);
+        ModelNode item = new ModelNode();
+        item.get(ATTR_1.getName()).set("newvalue");
+        item.get(ATTR_2.getName()).set(true);
+        ModelNode mapValue = item.get(MAP_ATTRIBUTE.getName());
+        mapValue.get("key1").set("value1");
+        mapValue.get("key2").set("value2");
+        op = createOperation("list-add", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName());
+        op.get(VALUE).set(item);
+        executeCheckNoFailure(op);
+        op = createOperation("read-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName() + "[0].attr1");
+        Assert.assertEquals("newvalue", executeForResult(op).asString());
+
+        // test write to non-existing attribute
+        op = createOperation("undefine-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName());
+        executeCheckNoFailure(op);
+        op = createOperation("write-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName());
+        ModelNode list = new ModelNode();
+        list.add(item);
+        op.get("value").set(list);
+        executeCheckNoFailure(op);
+
+        // test add to undefined parent
+        op = createOperation("undefine-attribute", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName());
+        executeCheckNoFailure(op);
+        item = new ModelNode();
+        item.get(ATTR_1.getName()).set("newvalue");
+        item.get(ATTR_2.getName()).set(true);
+        mapValue = item.get(MAP_ATTRIBUTE.getName());
+        mapValue.get("key1").set("value1");
+        mapValue.get("key2").set("value2");
+        op = createOperation("list-add", TEST_ADDRESS);
+        op.get("name").set(COMPLEX_ATTRIBUTE2.getName() + "." + OBJECT_LIST.getName());
+        op.get(VALUE).set(item);
+        executeCheckForFailure(op);  // TODO perhaps this could be made to work and this line would be replaced by the following
```

Comment:
why should this fail?
write operations of which list-add is one off, should be able to write whole structure.
That is one of the useful features of this syntax